### PR TITLE
Update Portal Auth flow

### PR DESCRIPTION
**Pull request by mheap**
### Summary
Update Portal Auth flow with the following changes:

* Setting a password now occurs after email validation
* Add more information about manual/automatic approval of developers
* Update screenshots to the new design
* Note that if you try to register multiple times, you'll get an email confirming the status of your account

There are copy changes I'd make to this section, but in the interests of small PRs I've left as much of the existing content as I could as-is. Once this PR is merged we can discuss reworking the text as a standalone PR.

### Reason
New release

### Testing

https://deploy-preview-3791--kongdocs.netlify.app/konnect/dev-portal/access-and-approval/dev-reg/
https://deploy-preview-3791--kongdocs.netlify.app/konnect/dev-portal/access-and-approval/manage-devs/

### CODE REVIEW

File: app/konnect/dev-portal/access-and-approval/dev-reg.md
Comment (lena-larionova) [inline app/konnect/dev-portal/access-and-approval/dev-reg.md:37]:
```suggestion
    After you have confirmed your email address and set a password, you’ll be able to log into the Dev Portal once your account admin has approved your request. 
```
Moving this sentence to previous step, as it sounds out of order.
Normally I'd shy away from future tense, but I think in this case it makes sense, as you might be waiting for approval.

File: app/konnect/dev-portal/access-and-approval/dev-reg.md
Comment (lena-larionova) [inline app/konnect/dev-portal/access-and-approval/dev-reg.md:33]:
```suggestion
4. You should receive an email to confirm your email address and set a password via the link inside the email. Click the verification link in the email.
```
Moved from step 5.

File: app/konnect/dev-portal/access-and-approval/dev-reg.md
Comment (lena-larionova) [inline app/konnect/dev-portal/access-and-approval/dev-reg.md:41]:
```suggestion
If your admin has auto-approve enabled, you can log into the Dev Portal immediately after setting a password. Learn how to [set up auto-approval for developer access as an admin](/konnect/dev-portal/access-and-approval/auto-approve-devs-apps/).
```

File: app/konnect/dev-portal/access-and-approval/dev-reg.md
Comment (lena-larionova) [inline app/konnect/dev-portal/access-and-approval/dev-reg.md:43]:
```suggestion
If auto-approve is not enabled, an admin has to review and approve your request manually. All manually approved developers are notified via email upon gaining access to the Dev Portal. Learn how to [manually approve developer access as an admin](/konnect/dev-portal/access-and-approval/manage-devs/#approve-dev-reg).
```

**Review by lena-larionova — CHANGES_REQUESTED** [review]

File: app/konnect/dev-portal/access-and-approval/dev-reg.md
Comment (lena-larionova) [inline app/konnect/dev-portal/access-and-approval/dev-reg.md:33]:
```suggestion
   You should receive an email to confirm your email address. Click the verification link in the email.
```

File: app/konnect/dev-portal/access-and-approval/dev-reg.md
Comment (lena-larionova) [inline app/konnect/dev-portal/access-and-approval/dev-reg.md:35]:
```suggestion
5. Set a password for your account.
```
Missed this. It's implied that it's after approval, since it's the next step.

**Review by lena-larionova — CHANGES_REQUESTED** [review]

**Review by lena-larionova — APPROVED** [review]

**Comment by lena-larionova** [comment]
@mheap I can't believe I just noticed this - this PR is open against `main`, but it should be targeting `konnect`. Can you please rebase and retarget?

**Comment by mheap** [comment]
@lena-larionova Now targeting `konnect`

Once @sichvoge reviews I'll hit merge

**Comment by sichvoge** [comment]
Only a small request to fix, but otherwise LGTM. We also found a potential bug as I went through the docs, trying out whats written there :) So thanks! :D

**Review by sichvoge — CHANGES_REQUESTED** [review]

File: app/konnect/dev-portal/access-and-approval/manage-devs.md
Comment (mheap) [reply app/konnect/dev-portal/access-and-approval/manage-devs.md:20]:
```suggestion
**Approved**
```

**Pull request merged**
